tests/test_plugin_loader.py

```
@@ -2310,6 +2310,351 @@ def test_clone_or_update_repo_checkout_fails_fallback(
+        self.assertFalse(result)
+        mock_logger.exception.assert_called_once()
+
+    def test_load_plugins_from_directory_auto_install_uses_pipx_inject(self):
+        """Missing deps should use pipx inject when running inside pipx."""
+        plugin_file = os.path.join(self.temp_plugins_dir, "pipx_plugin.py")
+        with open(plugin_file, "w", encoding="utf-8") as handle:
+            handle.write("import missingdep_pipx\n\nclass Plugin:\n    pass\n")
+
+        for var in ("PIPX_HOME", "PIPX_LOCAL_VENVS"):
+            os.environ.pop(var, None)
+        os.environ["PIPX_HOME"] = "/tmp/pipx-home"
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/pipx-home"

(S108)